jgrapht-core/src/test/java/org/jgrapht/traverse/DegeneracyOrderingIteratorTest.java

```diff
@@ -0,0 +1,199 @@
+import org.jgrapht.event.EdgeTraversalEvent;
+import org.jgrapht.event.TraversalListener;
+import org.jgrapht.event.VertexTraversalEvent;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.SimpleGraph;
+import org.junit.Test;
+
+/**
+ * Tests for {@link DegeneracyOrderingIterator}.
+ *
+ * @author Dimitrios Michail
+ */
+public class DegeneracyOrderingIteratorTest
+{
+    @Test
+    public void testGraph1()
+    {
+        SimpleGraph<String, DefaultEdge> g = new SimpleGraph<>(DefaultEdge.class);
+
+        g.addVertex("v1");
+        g.addVertex("v2");
+        g.addVertex("v3");
+        g.addVertex("v4");
+        g.addVertex("v5");
+        g.addVertex("v6");
+        g.addVertex("v7");
+        g.addVertex("v8");
+        g.addVertex("v9");
+        g.addVertex("v10");
+
+        // biggest clique: { V1, V2, V3, V4 }
+        g.addEdge("v1", "v2");
+        g.addEdge("v1", "v3");
+        g.addEdge("v1", "v4");
+        g.addEdge("v2", "v3");
+        g.addEdge("v2", "v4");
+        g.addEdge("v3", "v4");
+
+        // smaller clique: { V5, V6, V7 }
+        g.addEdge("v5", "v6");
+        g.addEdge("v5", "v7");
+        g.addEdge("v6", "v7");
+
+        // for fun, add an overlapping clique { V3, V4, V5 }
+        g.addEdge("v3", "v5");
+        g.addEdge("v4", "v5");
+
+        // make V8 less lonely
+        g.addEdge("v7", "v8");
+
+        // add one more maximal which is also the biggest { V1, V2, V9, V10 }
+        g.addEdge("v1", "v9");
+        g.addEdge("v1", "v10");
+        g.addEdge("v2", "v9");
+        g.addEdge("v2", "v10");
+        g.addEdge("v9", "v10");
+
+        StringBuffer sb = new StringBuffer();
+        DegeneracyOrderingIterator<String, DefaultEdge> it = new DegeneracyOrderingIterator<>(g);
+        while (it.hasNext()) {
+            String v = it.next();
+            sb.append("," + v);
+        }
+        assertEquals(",v8,v6,v7,v5,v9,v10,v1,v2,v3,v4", sb.toString());
+    }
+
+    @Test
+    public void testGraph2()
+    {
+        SimpleGraph<String, DefaultEdge> g = new SimpleGraph<>(DefaultEdge.class);
+
+        Graphs.addAllVertices(
+            g, Arrays.asList("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k"));
+        g.addEdge("b", "c");
+        g.addEdge("c", "d");
+        g.addEdge("c", "e");
+        g.addEdge("e", "f");
+        g.addEdge("e", "g");
+        g.addEdge("e", "h");
+        g.addEdge("f", "g");
+        g.addEdge("f", "h");
+        g.addEdge("f", "i");
+        g.addEdge("g", "h");
+        g.addEdge("i", "j");
+        g.addEdge("i", "k");
+        g.addEdge("j", "k");
+
+        StringBuffer sb = new StringBuffer();
+        DegeneracyOrderingIterator<String, DefaultEdge> it = new DegeneracyOrderingIterator<>(g);
+        while (it.hasNext()) {
+            String v = it.next();
+            sb.append("," + v);
+        }
+        assertEquals(",a,b,d,c,j,i,k,e,f,g,h", sb.toString());
+    }
+
+    @Test
+    public void testGraph2WithListener()
```

Comment:
This test seems pretty much the same as the `testGraph2` test, except that you added a listener. I would recommend combining the two.